### DISCUSSION TITLE
Visualize situations where the parameters of dimensionality reduction are inconsistent with those of the passed model

### DISCUSSION THREAD

**Discussion by superseanyoung** [MaartenGr/BERTopic#2216] [General]
When I use UAMP to reduce dimensions, I set n_neighbors=30,n_components=15, and then pass in BERTopic() with this UMAP. However, I want to use the visualize_document_datamap () method in the subsequent normalization, and then I introduce the reduced_embeddings, which I obtained through UMAP ().fit_transform. But the parameters I set there are n_neighbors=30,n_components=2, which is different from the parameters I passed into the BERTopic model, so what's the impact of that?

**Comment by MaartenGr** [comment]
There's always some mismatch between the actual procedure of fitting the model and the 2D representation since we cannot easily visualize in higher dimensionality than that. 2D visualizations are always an approximation of reality. That said, are you referring to something specific when you mention *"impact"*?

**Reply by superseanyoung** [reply]
My front and back parameters are different. Is there a logical error in this operation?Or maybe I can have the input model's parameters different from the parameters of my dimensionality reduction?

**Reply by MaartenGr** [reply]
> My front and back parameters are different. Is there a logical error in this operation?

I don't think so. As mentioned a 2D representation is still a proxy of the original, so take that for what it is. 

> Or maybe I can have the input model's parameters different from the parameters of my dimensionality reduction?

The 2D visualization is an approximation of the underlying clustering process. As such, you will never get the exact same thing. That is something to take into account, but you for example can't just use 15 components for both the fitting and the visualization as 15 dimensions cannot be visualized that way.

Therefore, you take 2D representations but the underlying principle is the same, albeit with information loss. This is the suggested approach and although it is not an exact match (and unlikely ever will), I would not be worried about the difference in parameters here other than what I mentioned above.

**Reply by SongLin99** [reply]
Hello bro,

It's so lucky that you have the same question as I do and that you posted it. After reading your conversation, I believe I now understand the problem.

Firstly, we use UMAP (n_neighbors=30, n_components=15) to reduce the dimensions and provide the reduced embeddings to HDBSCAN for clustering. After that, each embedding is assigned a category. At this point, clustering is complete. We then use UMAP (n_neighbors=30, n_components=2) to visualize the embeddings we obtained.

UMAP (n_neighbors=30, n_components=2) only works as a visualization tool. Because clustering is already done before.

**Reply by superseanyoung** [reply]
Which umap is used for two-dimensional visualization at this time? I mean, can the previous 15-dimensional clustering be displayed in the 2-dimensional space? Why

**Reply by SongLin99** [reply]
UMAP（n_neighbors=30，n_components=2）is used for two-dimensional visualization at this time. 15-dimensional can`t be understand by human, so we use 2-dimension as a _**approximate substitute**_, although it cannot completely replace the 15 dimension, it can roughly check the clustering effect
